filters/colorinterp/ColorinterpFilter.cpp

```diff
@@ -0,0 +1,221 @@
+{
+    // If the user] selected a default ramp name, it will be opened by
+    // one of these macros if it matches. Otherwise, we just open with the
+    // GDALOpen'able the user gave us
+
+    // GETRAMP will set rampFilename to the /vsimem filename it
+    // used to actually open the file, and from then on it can be treated
+    // like any other GDAL datasource.
+
+    GETRAMP(awesome_green);
+    GETRAMP(black_orange);
+    GETRAMP(blue_hue);
+    GETRAMP(blue_red);
+    GETRAMP(heat_map);
+    GETRAMP(pestel_shades);
+
+    std::shared_ptr<pdal::gdal::Raster> output (new pdal::gdal::Raster(rampFilename.c_str()));
+    return output;
+}
+
+void ColorinterpFilter::addArgs(ProgramArgs& args)
+{
+    std::string dimension;
+    args.add("dimension", "Dimension to interpolate", dimension, "Z");
+    args.add("minimum", "Minimum value to use for scaling", m_min);
+    args.add("maximum", "Maximum value to use for scaling", m_max);
+    args.add("ramp", "GDAL-readable color ramp image to use", m_colorramp, "pestel_shades");
+    args.add("invert", "Invert the ramp direction", m_invertRamp, false);
+    args.add("stddev", "Compute minimum/maximum given this many standard deviations", m_stdDevThreshold, 0.0);
+    m_interpDim = Dimension::id(dimension);
+}
+
+void ColorinterpFilter::addDimensions(PointLayoutPtr layout)
+{
+    layout->registerOrAssignDim("Red",
+        Dimension::defaultType(Dimension::Id::Red));
+    layout->registerOrAssignDim("Green",
+        Dimension::defaultType(Dimension::Id::Green));
+    layout->registerOrAssignDim("Blue",
+        Dimension::defaultType(Dimension::Id::Blue));
+}
+
+void ColorinterpFilter::initialize()
+{
+    gdal::registerDrivers();
+
+    m_raster = openRamp(m_colorramp);
+    m_raster->open();
+
+    log()->get(LogLevel::Debug) << getName() << " raster connection: "
+                                             << m_raster->m_filename << std::endl;
+}
+
+bool ColorinterpFilter::processOne(PointRef& point)
+{
+
+    double v = point.getFieldAs<double>(m_interpDim);
+    double red(1.0), green(1.0), blue(1.0);
+
+    size_t img_width = m_redBand.size();
+    double factor = (v - m_min) / (m_max - m_min);
+
+    if (m_invertRamp)
+        factor = 1 - factor;
+
+    size_t position(std::floor(factor * img_width));
+
+    // Don't color points whose computed position
+    // would fall outside the image
+    if (position > img_width)
+    {
+        return false;
+    }
+
+    red = m_redBand[position];
+    blue = m_blueBand[position];
+    green = m_greenBand[position];
+
+    point.setField(Dimension::Id::Red, red);
+    point.setField(Dimension::Id::Green, green);
+    point.setField(Dimension::Id::Blue, blue);
+
+    return false;
+}
+
+void ColorinterpFilter::filter(PointView& view)
+{
+    if ( m_stdDevThreshold != 0.0)
+    {
+        std::vector<double> values(view.size());
+
+        pdal::stats::Summary summary(pdal::Dimension::name(m_interpDim), pdal::stats::Summary::NoEnum);
```

Comment:
Seems like if we're going to use this outside of the stats filter, we should hoist it somewhere else.  We could just add the median thing, too.  Would make this a lot cleaner and wouldn't have to double-store the values.